pkg/controller/machine/controller.go

```diff
@@ -216,7 +217,18 @@ func (c *ReconcileMachine) delete(ctx context.Context, machine *clusterv1.Machin
 
 func (c *ReconcileMachine) getCluster(ctx context.Context, machine *clusterv1.Machine) (*clusterv1.Cluster, error) {
 	clusterList := clusterv1.ClusterList{}
-	err := c.Client.List(ctx, client.InNamespace(machine.Namespace), &clusterList)
+	listOptions := &client.ListOptions{
+		Namespace: machine.Namespace,
+		// This is set so the fake client can be used for unit test. See:
+		// https://github.com/kubernetes-sigs/controller-runtime/issues/168
+		Raw: &metav1.ListOptions{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: clusterv1.SchemeGroupVersion.String(),
+				Kind:       "Cluster",
+			},
+		},
+	}
+	err := c.Client.List(ctx, listOptions, &clusterList)
```

Comment:
nit: This should be cuddled, e.g. `if err := c.Client.List(...); err != nil { ... }`